### PR TITLE
Feat/ping 2998

### PR DESCRIPTION
**Pull request by iwangsyah**
### Summary by CodeRabbit

- Bug Fix: Adjusted the ScrollView's bottom margin in Topics and Who to Follow screens to prevent content from being hidden behind the device's bottom safe area.
- Refactor: Optimized the animation logic for displaying follow icons in the Who to Follow screen, improving the performance and visual experience.
- Style: Updated the height of the ItemUser component in the Who to Follow screen for better layout consistency. Removed unused styles for cleaner codebase.

### CODE REVIEW

**Comment by n-permana** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between beebe3a2543fb4ea39097c09f6037f25a4ecf393 and ea557106ffe33f99420ff09a22be330a32d4fdd9 commits.
</details>

<details>
<summary>Files selected (3)</summary>

* src/screens/Topics/index.js (4)
* src/screens/WhotoFollow/elements/ItemUser.js (5)
* src/screens/WhotoFollow/index.js (8)
</details>






---

File: src/screens/WhotoFollow/elements/ItemUser.js [hunk 1]
```diff
@@ -52,10 +47,10 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
     alignItems: 'center',
     justifyContent: 'space-between',
-    height: 76,
+    height: dimen.normalizeDimen(64),
     paddingHorizontal: 20,
     width: '100%',
-    flex: 1,
+    flex: 1
```
Comment (n-permana) [inline src/screens/WhotoFollow/elements/ItemUser.js:53]: Image description   CodeRabbit

The height of the container has been changed from a static value to a dynamic one using the `dimen.normalizeDimen` function. This is a good practice as it makes the UI more responsive and adaptable to different screen sizes.
```diff
-    height: 76,
+    height: dimen.normalizeDimen(64),
```

File: src/screens/Topics/index.js [hunk 2]
```diff
@@ -187,7 +189,9 @@ const Topics = () => {
   );
 };
 const styles = StyleSheet.create({
-  scrollViewStyle: {marginBottom: 100},
+  scrollViewStyle: (bottom) => ({
+    marginBottom: 100 - bottom
+  }),
```
Comment (n-permana) [inline src/screens/Topics/index.js:194]: Image description   CodeRabbit

The calculation for `marginBottom` seems to be incorrect. If `bottom` is the safe area inset at the bottom of the screen, then subtracting it from 100 might not give the desired margin. The `marginBottom` should be equal to `bottom` if you want the ScrollView content to be above the safe area.
```diff
- marginBottom: 100 - bottom
+ marginBottom: bottom
```

**Review by n-permana — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between beebe3a2543fb4ea39097c09f6037f25a4ecf393 and ea557106ffe33f99420ff09a22be330a32d4fdd9 commits.
</details>

<details>
<summary>Files selected (3)</summary>

* src/screens/Topics/index.js (4)
* src/screens/WhotoFollow/elements/ItemUser.js (5)
* src/screens/WhotoFollow/index.js (8)
</details>







<details>
<summary>Files skipped from review due to trivial changes (1)</summary>

* src/screens/WhotoFollow/index.js

</details>

<details>
<summary>Review comments generated (9)</summary>

* Review: 2
* LGTM: 7

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

File: src/screens/Topics/index.js (hunk 2 above)
Comment (iwangsyah) [reply src/screens/Topics/index.js:194]: need to add 100 because of button bottom height is 100

File: src/screens/Topics/index.js [hunk 3]
```diff
@@ -187,7 +189,9 @@ const Topics = () => {
   );
 };
 const styles = StyleSheet.create({
-  scrollViewStyle: {marginBottom: 100},
+  scrollViewStyle: (bottom) => ({
+    marginBottom: 100 - bottom
```
Comment (fajarism) [inline src/screens/Topics/index.js:193]: need to normalizeDimen?

File: src/screens/WhotoFollow/elements/ItemUser.js [hunk 4]
```diff
@@ -35,8 +31,7 @@ const ItemUser = ({photo, username, bio, followed, onPress, userid}) => {
           <Animated.View style={{position: 'absolute', opacity: 1}}>
             <IconAdd width={20} height={20} fill="#000000" />
```
Comment (fajarism) [inline src/screens/WhotoFollow/elements/ItemUser.js:32]: need to normalizeDimen?

File: src/screens/WhotoFollow/elements/ItemUser.js [hunk 5]
```diff
@@ -35,8 +31,7 @@ const ItemUser = ({photo, username, bio, followed, onPress, userid}) => {
...
           </Animated.View>
-          <Animated.View
-            style={{position: 'absolute', opacity: followIconFadeAnimation}}>
+          <Animated.View style={{position: 'absolute', opacity: followIconFadeAnimation}}>
             <IconCheck width={32} height={32} fill="#23C5B6" />
```
Comment (fajarism) [inline src/screens/WhotoFollow/elements/ItemUser.js:35]: need to normalizeDimen?

File: src/screens/WhotoFollow/elements/ItemUser.js [hunk 6]
```diff
@@ -52,10 +47,10 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
     alignItems: 'center',
     justifyContent: 'space-between',
-    height: 76,
+    height: dimen.normalizeDimen(64),
     paddingHorizontal: 20,
```
Comment (fajarism) [inline src/screens/WhotoFollow/elements/ItemUser.js:51]: need to normalizeDimen?

File: src/screens/WhotoFollow/index.js [hunk 7]
```diff
@@ -303,9 +306,9 @@ const styles = StyleSheet.create({
     flex: 1,
     backgroundColor: colors.white
   },
-  recyclerview: {
-    marginBottom: 112
-  },
+  recyclerview: (bottom) => ({
+    marginBottom: 112 - bottom
```
Comment (fajarism) [inline src/screens/WhotoFollow/index.js:310]: need to normalizeDimen?

File: src/screens/WhotoFollow/index.js [hunk 8]
```diff
@@ -334,7 +337,7 @@ const styles = StyleSheet.create({
     color: colors.gray,
     opacity: 0.84,
     marginTop: 8,
-    marginBottom: 24,
+    marginBottom: 12,
     paddingHorizontal: 20
```
Comment (fajarism) [inline src/screens/WhotoFollow/index.js:341]: need to normalizeDimen?

File: src/screens/WhotoFollow/index.js [hunk 9]
```diff
@@ -370,120 +373,6 @@ const styles = StyleSheet.create({
     fontSize: 10,
```
Comment (fajarism) [inline src/screens/WhotoFollow/index.js:373]: need to normalizeDimen?

**Review by fajarism — CHANGES_REQUESTED** [review]

**Pull request closed without merging**